### PR TITLE
Edit name command

### PR DESCRIPTION
**Pull request by ChaseTiong**
# Add a new command
- Edit Name command based on index

### CODE REVIEW

File: src/seedu/addressbook/commands/HelpCommand.java [hunk 1]
```diff
@@ -12,6 +12,7 @@
             + "Example: " + COMMAND_WORD;
 
     public static final String MESSAGE_ALL_USAGES = AddCommand.MESSAGE_USAGE
+    		+ "\n" + EditNameCommand.MESSAGE_USAGE
```
Comment (louietyj) [inline src/seedu/addressbook/commands/HelpCommand.java]: Indentation

File: src/seedu/addressbook/data/person/Person.java [hunk 2]
```diff
@@ -84,4 +85,9 @@ public String toString() {
         return getAsTextShowAll();
     }
 
+	public void setName(String newName) throws IllegalValueException {
+		this.name = new Name(newName);
+
```
Comment (louietyj) [inline src/seedu/addressbook/data/person/Person.java]: Spurious newlines

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 3]
```diff
@@ -131,4 +132,14 @@ public int hashCode() {
         return internalList.hashCode();
     }
 
+	public void editName(int index, String newName) throws 
+	PersonNotFoundException, IllegalValueException{
+		final int OFFSET = 1;
+		if(internalList.size() < index || index <= 0){
```
Comment (louietyj) [inline src/seedu/addressbook/data/person/UniquePersonList.java]: Missing spaces

File: src/seedu/addressbook/parser/Parser.java [hunk 4]
```diff
@@ -88,8 +95,39 @@ public Command parseCommand(String userInput) {
                 return new HelpCommand();
         }
     }
+    
+    private String trimSpace(String arguments){
+    	return arguments.trim();
+    }
+    
+    private String[] splitBySpace(String arguments) throws ParseException{
+    	if(trimSpace(arguments).split(SPACE).length <= 1){
```
Comment (louietyj) [inline src/seedu/addressbook/parser/Parser.java]: Same

File: src/seedu/addressbook/parser/Parser.java [hunk 5]
```diff
@@ -88,8 +95,39 @@ public Command parseCommand(String userInput) {
...
 
-    /**
+    /** 
+     * Parses arguments in the context of the edit person name command.
+     *
+     * @param argumentss full command arguments string
+     * @return the prepared command
+     */
+    
+    private Command prepareEdit(String arguments) {
```
Comment (louietyj) [inline src/seedu/addressbook/parser/Parser.java:118]: Indentation for whole block

**Review by louietyj — CHANGES_REQUESTED** [review]
Write some logic tests?

File: src/seedu/addressbook/commands/HelpCommand.java (hunk 1 above)
Comment (ChaseTiong) [reply src/seedu/addressbook/commands/HelpCommand.java]: fixed.

File: src/seedu/addressbook/data/person/Person.java (hunk 2 above)
Comment (ChaseTiong) [reply src/seedu/addressbook/data/person/Person.java]: fixed.

File: src/seedu/addressbook/data/person/UniquePersonList.java (hunk 3 above)
Comment (ChaseTiong) [reply src/seedu/addressbook/data/person/UniquePersonList.java]: fixed.

File: src/seedu/addressbook/parser/Parser.java (hunk 4 above)
Comment (ChaseTiong) [reply src/seedu/addressbook/parser/Parser.java]: fixed.

File: src/seedu/addressbook/parser/Parser.java (hunk 5 above)
Comment (ChaseTiong) [reply src/seedu/addressbook/parser/Parser.java:118]: fixed.

File: src/seedu/addressbook/parser/Parser.java [hunk 6]
```diff
@@ -88,14 +95,45 @@ public Command parseCommand(String userInput) {
...
+        try{
+		    final int chosenIndex = parseArgsAsDisplayedIndex(splitBySpace(arguments)[GET_INDEX]);
+			final String givenNewName = splitBySpace(arguments)[GET_NAME];
+			return new EditNameCommand(chosenIndex, givenNewName);
+		}
+		catch (ParseException | NumberFormatException e) {
+		    return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, 
+		            EditNameCommand.MESSAGE_USAGE));
+		}
+	}
```
Comment (louietyj) [inline src/seedu/addressbook/parser/Parser.java:128]: Lines 119-127 still not fixed.

**Review by louietyj — CHANGES_REQUESTED** [review]
Still pending linting errors.

**Comment by louietyj** [comment]
Wait is addressbook3 gonna be the base for our final Phase C submission? If not screw it just merge.

**Comment by louietyj** [comment]
Okay apparently not =x

**Pull request merged**